applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -127,10 +127,10 @@ int KratosGeoSettlement::RunStage(const std::filesystem::path&            rWorki
         std::vector<std::shared_ptr<Process>> processes = GetProcesses(project_parameters);
         std::vector<std::weak_ptr<Process>> process_observables(processes.begin(), processes.end());
 
-        if (mpTimeLoopExecutor)
-        {
-            mpTimeLoopExecutor->SetProcessReferences(process_observables);
-        }
+    if (mpTimeLoopExecutor)
+    {
+        mpTimeLoopExecutor->SetProcessObservables(process_observables);
+    }
```

Comment:
Nitpicking: by looking at the context, the indentation doesn't seem to be correct.
```suggestion
        if (mpTimeLoopExecutor)
        {
            mpTimeLoopExecutor->SetProcessObservables(process_observables);
        }
```

Also (for later, when we start to compose things) I feel that we must have a time loop executor, since without it there is nothing useful that we can do. Therefore, the constructor of this class should check whether the smart pointer is not null. Our class invariant then becomes we can safely assume that the pointer has been set (no further checks needed).

Reply:
Done ( the indentation )